build_tools/new_package.sh

```diff
@@ -10,57 +10,114 @@ ask() {
 	else
 		if echo "${1##*/}" | ${SED} 's/-//g' | ${SED} 's/\.tar.*//g' | ${SED} "s/${2}//g" | grep "${3}" &>/dev/null; then
-			echo -e "\twget -q -nc -P\$(BUILD_SOURCE) $(${SED} "s/${2}/\$(${4}_VERSION)/g" <<< "$1")"
+			echo -e "\twget -q -nc -P\$(BUILD_SOURCE) $(${SED} "s/${2}/\$(${4}_VERSION)/g" <<<"$1")"
 		else
 			echo -e "\t-[ ! -f "$\(BUILD_SOURCE\)/${3}-$\(${4}_VERSION\).tar.${download##*.}" ] \&\& \\
 \t	\twget -q -nc -O\$(BUILD_SOURCE)/${3}-\$(${4}_VERSION).tar.${download##*.}) \\
-\t	\t\t$(${SED} "s/${2}/\$(${4}_VERSION)/g" <<< "$1")"
+\t	\t\t$(${SED} "s/${2}/\$(${4}_VERSION)/g" <<<"$1")"
 		fi
 	fi
 }
 
 main() {
 	pkg="$(ask "Package Name" $1)"
-	checkpkg "$pkg"
-	formatpkg="$(${SED} -e 's|-|_|g' -e "s|\(.\)|\u\1|g" <<< "${pkg}")"
-	build="$(ask "Build System" $2)"
-	checkbuild "$build"
+	checkpkg "$pkg" || exit 1
+	formatpkg="$(${SED} -e 's|-|_|g' -e "s|\(.\)|\u\1|g" <<<"${pkg}")"
+
+	while true; do
+		build="$(ask "Build System (type ? to list all build systems)" $2)"
+		if [ "$build" = "?" ]; then
+			echo "Available build systems:"
+			find build_misc/templates -name '*.mk' -not -name 'keyring.mk' -exec basename '{}' .mk \; | $SED 's/^/- /g'
+		else
+			if checkbuild "$build"; then
+				break
+			elif ! [ -z "$2" ]; then
+				exit 1
+			fi
+		fi
+	done
+
 	ver="$(ask "Package Version" $3)"
 	download="$(ask "Tarball Download Link" $4)"
 	${SED} -e "s/@pkg@/${pkg}/g" \
 		-e "s/@PKG@/${formatpkg}/g" \
 		-e "s/@PKG_VERSION@/${ver}/g" \
 		-e "s|@download@|$(downloadlink "$download" "$ver" "$pkg" "$formatpkg")|g" \
 		-e "s|@compression@|${download##*.}|g" \
-		build_misc/templates/${build}.mk > makefiles/${pkg}.mk
+		"build_misc/templates/${build}.mk" > "makefiles/${pkg}.mk"
 }
 
-if which gsed &>/dev/null; then
+main_dialog() {
+	pkg=$(dialog_inputbox "New package" "Package name")
+	if ! checkpkg "$pkg"; then
+		dialog --msgbox "$pkg already exists" 15 40
+		clear
+		exit 1
+	fi
+	formatpkg="$(${SED} -e 's|-|_|g' -e "s|\(.\)|\u\1|g" <<<"${pkg}")"
+
+	buildsystems=()
+	count=1
+	for i in ./build_misc/templates/*.mk; do
+		if ! [ "$(basename $i .mk)" = "keyring" ]; then
+			buildsystems+=($count $(basename $i .mk))
+			count=$((count + 1))
+		fi
+	done
+	build=$(dialog \
+		--colors --no-cancel --title 'New package' \
+		--menu 'Build System' 15 40 $count "${buildsystems[@]}" 3>&1 1>&2 2>&3 3>&-)
+	build=${buildsystems[(( build*2-1 ))]}
+
+	ver=$(dialog_inputbox "New package" "Package version")
+	download=$(dialog_inputbox "New package" "Tarball download link")
+	${SED} -e "s/@pkg@/${pkg}/g" \
+		-e "s/@PKG@/${formatpkg}/g" \
+		-e "s/@PKG_VERSION@/${ver}/g" \
+		-e "s|@download@|$(downloadlink "$download" "$ver" "$pkg" "$formatpkg")|g" \
+		-e "s|@compression@|${download##*.}|g" \
+		"build_misc/templates/${build}.mk" > "makefiles/${pkg}.mk"
+	clear
+}
+
+if command -v gsed &>/dev/null; then
 	SED=gsed
 else
+	# shellcheck disable=SC2209
```

Comment:
You should remove this ([#L114](https://github.com/ProcursusTeam/Procursus/blob/4649af1e7a8e8d76b919a5ea1b3e17dd73b49c07/build_tools/new_package.sh#L114))

Reply:
That doesn't really matter imo.